src/server/fedavg.py

```diff
@@ -218,6 +218,37 @@ def __init__(
         if use_fedavg_client_cls:
             self.init_trainer()
 
+        # create setup for centralized evaluation
+        self.dataset = self.get_dataset()
+        data_indices = self.get_clients_data_indices()
+        val_indices = np.concatenate(
+            [data_indices[i]["val"] for i in self.train_clients]
+        )
+        test_indices = np.concatenate(
+            [data_indices[i]["test"] for i in self.train_clients]
+        )
+        train_indices = np.concatenate(
+            [data_indices[i]["train"] for i in self.train_clients]
+        )
+        self.valset = torch.utils.data.Subset(self.dataset, val_indices)
+        self.testset = torch.utils.data.Subset(self.dataset, test_indices)
+        self.trainset = torch.utils.data.Subset(self.dataset, train_indices)
+        self.valloader = torch.utils.data.DataLoader(
+            self.valset,
+            batch_size=self.args.common.batch_size,
+            shuffle=False,
+        )
+        self.testloader = torch.utils.data.DataLoader(
+            self.testset,
+            batch_size=self.args.common.batch_size,
+            shuffle=False,
+        )
+        self.trainloader = torch.utils.data.DataLoader(
+            self.trainset,
+            batch_size=self.args.common.batch_size,
+            shuffle=False,
+        )
+
```

Comment:
I think we should wrap them into a function and add control codes to ignore  it if the central evalutation feature is not activated.